src/utils/parse_utils.ts

```diff
@@ -10,7 +11,101 @@ import {
+    Uniswap: true,
+    Balancer: true,
+    Curve: true,
+    Eth2Dai: true,
+    Kyber: true,
+    LiquidityProvider: true,
+    MultiBridge: true,
+    Uniswap_V2: true,
+};
+
 export const parseUtils = {
+    parseRequestForExcludedSources(
+        request: SwapRequestParams,
+        validApiKeys: string[],
+    ): { excludedSources: ERC20BridgeSource[]; nativeExclusivelyRFQT: boolean } {
+        // Ensure that both filtering arguments cannot be present.
+        if (request.excludedSources !== undefined && request.includedSources !== undefined) {
+            throw new ValidationError([
+                {
+                    field: 'excludedSources',
+                    code: ValidationErrorCodes.IncorrectFormat,
+                    reason: ValidationErrorReasons.ConflictingFilteringArguments,
+                },
+                {
+                    field: 'includedSources',
+                    code: ValidationErrorCodes.IncorrectFormat,
+                    reason: ValidationErrorReasons.ConflictingFilteringArguments,
+                },
+            ]);
+        }
+
+        // If excludedSources is present, parse the string array and return
+        if (request.excludedSources !== undefined) {
+            return {
+                excludedSources: parseUtils.parseStringArrForERC20BridgeSources(request.excludedSources.split(',')),
+                nativeExclusivelyRFQT: false,
+            };
+        }
+
+        if (request.includedSources !== undefined) {
+            // Only RFQT is eligible as of now
+            if (request.includedSources !== 'RFQT') {
+                throw new ValidationError([
+                    {
+                        field: 'includedSources',
+                        code: ValidationErrorCodes.IncorrectFormat,
+                        reason: ValidationErrorReasons.ArgumentNotYetSupported,
+                    },
+                ]);
+            }
+
+            // We assume that if a `takerAddress` key is present, it's value was already validated by the JSON
+            // schema.
+            if (request.takerAddress === undefined || request.takerAddress.length === 0) {
+                throw new ValidationError([
+                    {
+                        field: 'takerAddress',
+                        code: ValidationErrorCodes.IncorrectFormat,
+                        reason: ValidationErrorReasons.FieldInvalid,
+                    },
+                ]);
+            }
+
+            // We enforce a valid API key - we don't want to fail silently.
+            if (!validApiKeys.includes(request.apiKey)) {
+                throw new ValidationError([
+                    {
+                        field: '0x-api-key',
+                        code: ValidationErrorCodes.IncorrectFormat,
+                        reason: ValidationErrorReasons.FieldInvalid,
+                    },
+                ]);
+            }
+            return {
```

Comment:
just to confirm here @steveklebanoff we shouldn't be making any check for `intentOnFilling` right? I partially remember our discussion and wanted to confirm :)

Reply:
Per convo, let's do it